src/Microsoft.AspNet.Authentication.Facebook/FacebookAuthenticationHandler.cs

```diff
@@ -49,8 +49,20 @@ public FacebookAuthenticationHandler(HttpClient httpClient)
             return new TokenResponse(response);
         }
 
-        protected override async Task<AuthenticationTicket> GetUserInformationAsync(AuthenticationProperties properties, TokenResponse tokens)
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(AuthenticationProperties properties, TokenResponse tokens)
         {
+            var ticket = await base.CreateTicketAsync(properties, tokens);
+            if (ticket == null)
+            {
+                return null;
+            }
+            
+            var identity = ticket.Principal?.Identity as ClaimsIdentity;
```

Comment:
I'm not a huge fan of that, but `AuthenticationTicket.Principal` is not guaranteed to be non-null and `ClaimsPrincipal.Identity` may be null when no identity has been added to the principal.

@Tratcher should we update `AuthenticationTicket` to avoid these 2 cases or is this null check fine?